### PR TITLE
feat: Content from base url

### PR DESCRIPTION
**Pull request by nirgur**
## Related Issues

Fixes https://github.com/descope/etc/issues/8253

### CODE REVIEW

File: packages/sdks/web-component/src/app/index.html [hunk 1]
```diff
@@ -39,12 +39,13 @@
     <script type="module" src="../lib/descope-wc/index.ts"></script>
     <div class="login-container">
       <descope-wc
-        project-id="<project-id>"
+        project-id="P2qLFlkB1yk929xihOImeIPcNBEq"
```
Comment (asafshen) [inline packages/sdks/web-component/src/app/index.html:42]: kind reminder to revert :-)

File: packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/fetchWithFallbacks.ts
Comment (asafshen) [inline packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/fetchWithFallbacks.ts:1]: great util that helps separation of concerns 👍

File: packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/fetchWithFallbacks.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  const fallbacksArr = Array.isArray(fallbacks) ? fallbacks : [fallbacks];
+
+  for (let index = 0; index < fallbacksArr.length; index++) {
+    const url = fallbacksArr[index];
+    const isLast = index === fallbacksArr.length - 1;
+
+    try {
+      const res = await fetch(url, init);
+      if (res.ok) {
+        onSuccess?.(index);
```
Comment (asafshen) [inline packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/fetchWithFallbacks.ts:23]: might be just a personal opinion, but I think that the fact that the `onSuccess` can be replaced with a complex response (`{ res: ReturnType<typeof fetch>;  urlIndex: number }`)

IMHO its simpler to return such type than having two ways to interact with this function 
(if we would use fetch and return value in different places that is something else, but its not the case here)

this is just my opinion ofc

File: packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/staticResourcesMixin.ts [hunk 3]
```diff
@@ -35,30 +36,94 @@ export const staticResourcesMixin = createSingletonMixin(
...
+        if (this.baseStaticUrl) {
+          return getResourceUrl({
+            projectId: this.projectId,
+            filename,
+            baseUrl: this.baseStaticUrl,
+          });
+        }
+
+        const isBaseUrlUpdated = this.#lastBaseUrl !== this.baseUrl;
+        const shouldFallbackFetch = isBaseUrlUpdated && !!this.baseUrl;
```
Comment (asafshen) [inline packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/staticResourcesMixin.ts:59]: is there a case where there is any empty baseUrl?

File: packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/staticResourcesMixin.ts [hunk 4]
```diff
@@ -35,30 +36,94 @@ export const staticResourcesMixin = createSingletonMixin(
...
+
+        // if there are multiple resource urls, it means that there are fallbacks,
+        // if one of the options (which is not the last) is working, we want to keep using it by updating the workingBaseUrl
+        const onSuccess = !Array.isArray(resourceUrls)
+          ? null
+          : (index: number) => {
+              if (index !== resourceUrls.length - 1) {
+                const { origin, pathname } = new URL(resourceUrls[index]);
+                this.#workingBaseUrl = new URL(
+                  pathname.replace(
```
Comment (asafshen) [inline packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/staticResourcesMixin.ts:104]: why do we need this replacement?

**Review by asafshen — COMMENTED** [review]
looks good 👍 
minor comments and questions, lets go over it together

File: packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/staticResourcesMixin.ts (hunk 3 above)
Comment (nirgur) [reply packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/staticResourcesMixin.ts:59]: fixed

File: packages/libs/sdk-mixins/package.json [hunk 5]
```diff
@@ -30,6 +30,16 @@
         "types": "./dist/index.d.ts",
         "default": "./dist/cjs/mixins/themeMixin/index.js"
       }
+    },
+    "./staticResourcesMixin": {
```
[the remark this answers is not in the document]
Comment (nirgur) [reply packages/libs/sdk-mixins/package.json:34]: fixed

File: packages/sdks/web-component/src/app/index.html (hunk 1 above)
Comment (nirgur) [reply packages/sdks/web-component/src/app/index.html:42]: fixed

File: packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/fetchWithFallbacks.ts (hunk 2 above)
Comment (nirgur) [reply packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/fetchWithFallbacks.ts:23]: Decided to keep the current impl

File: packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/staticResourcesMixin.ts (hunk 4 above)
Comment (nirgur) [reply packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/staticResourcesMixin.ts:104]: modified

**Review by asafshen — APPROVED** [review]

**Review by asafshen — DISMISSED** [review]

**Pull request merged**